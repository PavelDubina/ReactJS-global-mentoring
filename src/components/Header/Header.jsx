import React from 'react'
import Link from 'next/link'
import PropTypes from 'prop-types'
import styles from '../../../styles/Header.module.scss'

export const Header = ({ children }) => (
  <header className={styles.container}>
    <div className={styles.container_shadow} />
    <section className={styles.section}>
      <Link href="/">
        <span className={styles.logo}>
          <b>netflix</b>
          roulette
        </span>
      </Link>
      {children}
    </section>
  </header>
)

Header.propTypes = {
  children: PropTypes.node.isRequired,
}
