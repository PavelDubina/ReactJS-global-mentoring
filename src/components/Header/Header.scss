.container {
  width: 100%;
  position: relative;
  min-height: 360px;
  background-color: #000b;
  margin: 0 0 10px 0;
  padding: 0 48px;

  &_shadow {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    background-image: url('../../static/images/movies-background.jpg');
    background-size: cover;
    filter: blur(2px);
    z-index: -1;
  }
}

.section {
  height: 100%;
  position: relative;
  max-width: 1280px;
  margin: 0 auto;
  padding: 18px 0;
}

.logo {
  position: absolute;
  font-size: 22px;
  line-height: 22px;
  color: #f65261;
  cursor: pointer;
  z-index: 50;
}
