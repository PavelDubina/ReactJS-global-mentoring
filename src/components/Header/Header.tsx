import React from 'react'
import Link from 'next/link'
import className from 'classnames'
import { useRouter } from 'next/router'
import styles from '../../../styles/Header.module.scss'
import ruIcon from '../../../public/RUS.svg'
import enIcon from '../../../public/Britain.svg'

export type HeaderProps = {
  children: React.ReactNode
}

export const Header: React.FC<HeaderProps> = ({ children }) => {
  const router = useRouter()
  const { asPath } = router
  const enBtnStyle = className(styles.en, { [styles.locale_active]: router.locale === 'en' })
  const ruBtnStyle = className(styles.ru, { [styles.locale_active]: router.locale === 'ru' })
  return (
    <header className={styles.container}>
      <div className={styles.container_shadow} />
      <section className={styles.section}>
        <Link href="/">
          <span className={styles.logo}>
            <b>netflix</b>
            roulette
          </span>
        </Link>
        <Link href={asPath} locale="en">
          <img className={enBtnStyle} src={enIcon} alt="EN" />
        </Link>
        <Link href={asPath} locale="ru">
          <img className={ruBtnStyle} src={ruIcon} alt="RU" />
        </Link>
        {children}
      </section>
    </header>
  )
}
