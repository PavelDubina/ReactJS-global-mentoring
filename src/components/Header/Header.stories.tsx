import React from 'react'
import { Story, Meta } from '@storybook/react'
import { Header, HeaderProps } from './Header'
import { Search } from '../Search/Search'
import { MovieDetails } from '../MovieDetails/MovieDetails'

export default {
  title: 'MyStorybook/Header',
  component: Header,
} as Meta

const Template: Story<HeaderProps> = (args) => <Header {...args} />

export const withoutAnything = Template.bind({})
withoutAnything.args = {
  children: null,
}

export const withSearch = Template.bind({})
withSearch.args = {
  children: <Search />,
}

export const withMovieDetails = Template.bind({})
withMovieDetails.args = {
  children: <MovieDetails />,
}
