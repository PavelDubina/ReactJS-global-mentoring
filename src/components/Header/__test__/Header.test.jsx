import React from 'react'
import { render } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'
import { Header } from '../Header'

describe('Header', () => {
  const getComponent = () =>
    render(
      <Router>
        <Header>
          <div>Test children</div>
        </Header>
      </Router>,
    )
  it('Should renders correctly', () => {
    expect(getComponent({}).asFragment()).toMatchSnapshot()
  })
})
