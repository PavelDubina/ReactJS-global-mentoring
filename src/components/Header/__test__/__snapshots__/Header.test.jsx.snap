// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should renders correctly 1`] = `
<DocumentFragment>
  <header
    class="container"
  >
    <div
      class="container_shadow"
    />
    <section
      class="section"
    >
      <span
        class="logo"
      >
        <b>
          netflix
        </b>
        roulette
      </span>
      <div>
        Test children
      </div>
    </section>
  </header>
</DocumentFragment>
`;
